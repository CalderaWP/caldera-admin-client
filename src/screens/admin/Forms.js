import React from 'react';
import Toolbar from './Toolbar';
import Viewer from './Viewer';
import AdminSlot from "./AdminSlot";
import {FormList} from "../../components/FormsList/FormList";
import {Notice} from '@wordpress/components'

export default class Forms extends AdminSlot {

	getContent(forms) {
		if (0 !== Object.keys(forms).length) {
			return (
				<div>

						<FormList
							forms={forms}
							onFormUpdate={this.props.onFormUpdate}
							openEntryViewerForForm={this.props.onSelectForm}

						/>

				</div>
			);
		}
		return (
			<Notice
				status="error"
				isDismissible
			>
				No Forms Found
			</Notice>
		);
	}

	render() {
		const {forms} = this.props;
		return (
			<div>
				<Toolbar.NavBar
					label="Forms"
					onActive={this.handleActive}
					onDeactive={this.handleDeactive}
					isActive={this.state.active}
				/>
				{this.state.active &&
				<Viewer.Content>
					{this.getContent(forms)}
				</Viewer.Content>
				}

			</div>
		);
	}


}


Forms.propTypes = FormList.propTypes;
Forms.defaultProps = {
	forms: {},
	entries: {}
};