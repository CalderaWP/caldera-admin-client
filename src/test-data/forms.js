export  const formwithIdCf1 = {
	ID: 'CF1',
	name: 'name One',
	field_details: {
		"order": {
			"firstName": {
				"id": "firstName",
				"label": "First Name"
			},
			"lastName": {
				"id": "lastName",
				"label": "Last Name"
			},
			"strongName": {
				"id": "strongName",
				"label": "Strong Bad"
			}
		},
		"entry_list": {
			"id": {
				"id": "id",
				"label": "ID"
			},
			"datestamp": {
				"id": "datestamp",
				"label": "Submitted"
			}
		}
	}

};
export const formWithIdCf2 =  {
	"ID": "cf2",
	"name": "Primary Contact Form",
	"fields": {
		"fld_8768091": {
			"ID": "fld_8768091",
			"name": "First Name",
			"type": "text"
		},
		"fld_9970286": {
			"ID": "fld_9970286",
			"name": "Last Name",
			"type": "text"
		},
		"fld_6009157": {
			"ID": "fld_6009157",
			"name": "Email Address",
			"type": "email"
		},
		"fld_7683514": {
			"ID": "fld_7683514",
			"name": "Comments Questions",
			"type": "paragraph"
		}
	},
	"emailIdentifyingFields": [],
	"piiFields": [],
	"privacyExporterEnabled": false,
	field_details: {
		"order": {
			"fld_8768091": {
				"id": "fld_8768091",
				"label": "First Name"
			},
			"fld_9970286": {
				"id": "fld_9970286",
				"label": "Last Name"
			},
			"fld_6009157": {
				"id": "fld_6009157",
				"label": "Email Address"
			},
			"fld_7683514": {
				"id": "fld_7683514",
				"label": "Comments Questions"
			}
		},
		entry_list: {
			"id": {
				"id": "id",
				"label": "ID"
			},
			"datestamp": {
				"id": "datestamp",
				"label": "Submitted"
			}
		}
	},
	"mailer": {
		"active": false
	},
	"entries": {
		count: 3
	}
};
/**
 * Mock form collection.
 */
export default {
	cf1: formwithIdCf1,
	cf2: formWithIdCf2
};