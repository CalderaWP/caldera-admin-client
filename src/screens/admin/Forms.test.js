import renderer from 'react-test-renderer';
import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Forms from './Forms';
import forms from '../../test-data/forms'
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});
const handler = () => {
};


describe('Forms component', () => {
	it('Renders empty div when not active and has no forms', () => {
		expect(
			renderer.create(
				<Forms
					forms={{}}
					openEntryViewerForForm={handler}
					onFormUpdate={handler}
				/>
			).toJSON()
		).toMatchSnapshot();
	});

	it('Renders empty div when not active and has  forms', () => {
		expect(
			renderer.create(
				<Forms
					forms={forms}
					openEntryViewerForForm={handler}
					onFormUpdate={handler}
				/>
			).toJSON()
		).toMatchSnapshot();
	});
	it('Renders when active and has no forms', () => {
		const component = shallow(<Forms
			forms={{}}
			openEntryViewerForForm={handler}
			onFormUpdate={handler}
		/>);

		component.setState({active: true});
		expect(
			toJson(component)
		).toMatchSnapshot();
	});
	it('Renders when active and has forms', () => {
		const component = shallow(<Forms
			forms={forms}
			openEntryViewerForForm={handler}
			onFormUpdate={handler}
		/>);

		component.setState({active: true});
		expect(
			toJson(component)
		).toMatchSnapshot();
	});


});