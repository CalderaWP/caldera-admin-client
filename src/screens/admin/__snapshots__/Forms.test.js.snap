// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms component Renders empty div when not active and has  forms 1`] = `<div />`;

exports[`Forms component Renders empty div when not active and has no forms 1`] = `<div />`;

exports[`Forms component Renders when active and has forms 1`] = `
<div>
  <Component
    isActive={true}
    label="Forms"
    onActive={[Function]}
    onDeactive={[Function]}
  />
  <Component>
    <div>
      <Component
        forms={
          Object {
            "cf1": Object {
              "ID": "CF1",
              "field_details": Object {
                "entry_list": Object {
                  "datestamp": Object {
                    "id": "datestamp",
                    "label": "Submitted",
                  },
                  "id": Object {
                    "id": "id",
                    "label": "ID",
                  },
                },
                "order": Object {
                  "firstName": Object {
                    "id": "firstName",
                    "label": "First Name",
                  },
                  "lastName": Object {
                    "id": "lastName",
                    "label": "Last Name",
                  },
                  "strongName": Object {
                    "id": "strongName",
                    "label": "Strong Bad",
                  },
                },
              },
              "name": "name One",
            },
            "cf2": Object {
              "ID": "cf2",
              "emailIdentifyingFields": Array [],
              "entries": Object {
                "count": 3,
              },
              "field_details": Object {
                "entry_list": Object {
                  "datestamp": Object {
                    "id": "datestamp",
                    "label": "Submitted",
                  },
                  "id": Object {
                    "id": "id",
                    "label": "ID",
                  },
                },
                "order": Object {
                  "fld_6009157": Object {
                    "id": "fld_6009157",
                    "label": "Email Address",
                  },
                  "fld_7683514": Object {
                    "id": "fld_7683514",
                    "label": "Comments Questions",
                  },
                  "fld_8768091": Object {
                    "id": "fld_8768091",
                    "label": "First Name",
                  },
                  "fld_9970286": Object {
                    "id": "fld_9970286",
                    "label": "Last Name",
                  },
                },
              },
              "fields": Object {
                "fld_6009157": Object {
                  "ID": "fld_6009157",
                  "name": "Email Address",
                  "type": "email",
                },
                "fld_7683514": Object {
                  "ID": "fld_7683514",
                  "name": "Comments Questions",
                  "type": "paragraph",
                },
                "fld_8768091": Object {
                  "ID": "fld_8768091",
                  "name": "First Name",
                  "type": "text",
                },
                "fld_9970286": Object {
                  "ID": "fld_9970286",
                  "name": "Last Name",
                  "type": "text",
                },
              },
              "mailer": Object {
                "active": false,
              },
              "name": "Primary Contact Form",
              "piiFields": Array [],
              "privacyExporterEnabled": false,
            },
          }
        }
        onFormUpdate={[Function]}
      />
    </div>
  </Component>
</div>
`;

exports[`Forms component Renders when active and has no forms 1`] = `
<div>
  <Component
    isActive={true}
    label="Forms"
    onActive={[Function]}
    onDeactive={[Function]}
  />
  <Component>
    <Notice
      isDismissible={true}
      status="error"
    >
      No Forms Found
    </Notice>
  </Component>
</div>
`;
